### PR TITLE
Distances

### PR DESCRIPTION
**Pull request by khenderick**

### CODE REVIEW

File: ovs/dal/hybrids/storagedriver.py [hunk 1]
```diff
@@ -187,6 +189,28 @@ def _cluster_node_config(self):
...
+            if len(primary_domains) == 0:
+                distance_map[str(sd.storagedriver_id)] = StorageDriver.DISTANCES.NEAR
+            else:
+                distance = StorageDriver.DISTANCES.INFINITE
+                for junction in sd.storagerouter.domains:
+                    if junction.backup is False:
+                        if junction.domain_guid in secondary_domains:
+                            distance = min(distance, StorageDriver.DISTANCES.FAR)
+                        if junction.domain_guid in primary_domains:
+                            distance = min(distance, StorageDriver.DISTANCES.NEAR)
```
Comment (kvanhijf) [inline ovs/dal/hybrids/storagedriver.py:210]: Reversing these if statements and breaking when finding the NEARest, is more optimal

File: ovs/lib/storagedriver.py [hunk 2]
```diff
@@ -78,6 +80,72 @@ def volumedriver_error(code, volume_id):
...
+
+                current_configs = vpool.clusterregistry_client.get_node_configs()
+                changes = len(current_configs) == 0
+                node_configs = []
+                for sd in vpool.storagedrivers:
+                    sd.invalidate_dynamics(['cluster_node_config'])
+                    new_config = sd.cluster_node_config
+                    node_configs.append(ClusterNodeConfig(**new_config))
+                    current_node_configs = [config for config in current_configs if config.vrouter_id == sd.storagedriver_id]
+                    if len(current_node_configs) == 1:
```
Comment (kvanhijf) [inline ovs/lib/storagedriver.py]: The ```changes```  will not be updated when starting off with 0 ```current_configs```

File: ovs/lib/storagedriver.py [hunk 3]
```diff
@@ -78,6 +80,72 @@ def volumedriver_error(code, volume_id):
...
+                                    sd.guid, storagerouter.name
+                                ))
+                            else:
+                                StorageDriverController._logger.exception('Got exception when validating StorageDriver {0} on StorageRouter {1}.'.format(
+                                    sd.guid, storagerouter.name
+                                ))
+
+                    StorageDriverController._logger.info('Updating cluster node configs for VPool {0}'.format(vpool.guid))
+                    vpool.clusterregistry_client.set_node_configs(node_configs)
+                    srclient = StorageDriverClient.load(vpool)
```
Comment (kvanhijf) [inline ovs/lib/storagedriver.py]: More optimal to use ```vpool.storagedriver_client``` here

**Review by kvanhijf — CHANGES_REQUESTED** [review]

**Comment by khenderick** [comment]
Tests are not passing yet because we're missing the correct volumedriver package at this moment. I'll trigger rebuilds as soon as it's available.

**Review by kvanhijf — APPROVED** [review]

**Pull request merged**